clients/pkg/promtail/targets/file/decompresser.go

```diff
@@ -176,15 +190,19 @@ func (t *decompresser) readLines() {
 		return
 	}
 
-	level.Info(t.logger).Log("msg", "successfully decompressed file", "path", t.path)
+	level.Info(t.logger).Log("msg", "successfully mounted reader", "path", t.path, "ext", filepath.Ext(t.path))
 
 	scanner := bufio.NewScanner(r)
 	for i := 0; ; i++ {
 		if !scanner.Scan() {
 			break
 		}
 
-		if scanner.Err() != nil {
+		if scannerErr := scanner.Err(); scannerErr != nil {
+			if scannerErr != io.EOF {
+				level.Error(t.logger).Log("msg", "error scanning", "err", scannerErr)
```

Comment:
Will this message give us an indication of which target has a problem?

Reply:
hmmm you mean literally the target (which is the file)? If so, kinda; the component (which is the decompresser) is injected into the logger and by it you can infer that it is the file target.

Reply:
OK perfect, thanks